### PR TITLE
scripts: logging: dictionary: Add uart support

### PR DESCRIPTION
**Pull request by harish1992**
This commit adds support to dictionary logging to parse binary data directly from input serial and generate the ascii logs.

### CODE REVIEW

File: scripts/logging/dictionary/log_parser_uart.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        sys.exit(1)
+
+    logging.basicConfig(format=LOGGER_FORMAT)
+
+    if args.debug:
+        logger.setLevel(logging.DEBUG)
+    else:
+        logger.setLevel(logging.INFO)
+
+    # Parse the 2Kbs of log every second from uart
```
Comment (nordic-krch) [inline scripts/logging/dictionary/log_parser_uart.py:53]: 2k/sec seems small. I guess that we can easily get more data than that. Not sure what is better, increase it or decrease the sleep time.

File: scripts/logging/dictionary/log_parser_uart.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    with serial.Serial(args.serialPort, args.baudrate) as ser:
+        ser.timeout = 2
+        while True:
+            if ser.inWaiting():
+                data = ser.read(2048)
+                parser.parser(data, args.dbfile, logger)
+            else:
+                print('no data')
```
Comment (nordic-krch) [inline scripts/logging/dictionary/log_parser_uart.py:61]: i would remove that print.

Comment (harish1992) [reply scripts/logging/dictionary/log_parser_uart.py:61]: Will take care of this

**Review by nordic-krch — DISMISSED** [review]
lgtm

**Review by nordic-krch — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**